src/main/scala/is/hail/methods/FilterIntervals.scala

```diff
@@ -12,13 +12,13 @@ object FilterIntervals {
   def apply(vsm: MatrixTable, intervals: java.util.ArrayList[Interval], keep: Boolean): MatrixTable = {
     vsm.requireRowKeyVariant("filter_intervals")
     val locusField = vsm.rowType.fieldByName(vsm.rowKey(0))
-    val iList = IntervalTree(locusField.typ.ordering, intervals.asScala.toArray)
+    val iList = IntervalTree[Any](locusField.typ.ordering, intervals.asScala.toArray)
     apply(vsm, iList, keep)
   }
 
-  def apply[U](vsm: MatrixTable, intervals: IntervalTree[U], keep: Boolean): MatrixTable = {
+  def apply[U](vsm: MatrixTable, intervals: IntervalTree[Any, U], keep: Boolean): MatrixTable = {
     if (keep) {
-      val pkIntervals = IntervalTree(
+      val pkIntervals = IntervalTree[Any](
         vsm.matrixType.orvdType.pkType.ordering,
```

Comment:
OK, this code is kind of broken.  You didn't change it, but can we fix it up?  Namely, in the keep=true case, the interval tree gets rebuilt twice for no reason.  You should modify the apply-from-python to convert the ArrayList to an Array and then directly call the apply-from-scala, which should build the interval list in each case as needed.